project/BuildSettings.scala

```diff
@@ -14,6 +14,30 @@ import sbt._
 //// Basic settings
 ////*******************************
 object BuildSettings extends AutoPlugin {
+
+  // * Scalac Options
+  val scalacOptions212Only =
+    Seq(
+      "-Ywarn-adapted-args", // Warn if an argument list is modified to match the receiver.
+      "-Ywarn-inaccessible", // Warn about inaccessible types in method signatures.
+      "-Ywarn-nullary-override" // Warn when non-nullary overrides nullary, e.g. def foo() over def foo.
+    )
+
+  val scalacOptions213Only =
+    Seq()
+
+  val scalacOptionsCommon =
+    Seq(
+      "-deprecation", // Emit warning and location for usages of deprecated APIs.
+      "-feature", // Emit warning and location for usages of features that should be imported explicitly.
+      "-unchecked", // Enable additional warnings where generated code depends on assumptions.
+      //"-Xfatal-warnings", // Fail the compilation if there are any warnings.
```

Comment:
I guess this is commented out because it breaks the compilation, otherwise I would definitely have it in